### PR TITLE
Fix compilation and linking on arm platform (raspberry pi)

### LINKED ISSUE #1: Compilation errors on raspberry pi
https://mail.mozilla.org/pipermail/heka/2015-July/000655.html

### PR DESCRIPTION
**Pull request by bbinet**
This is based on @trink patch from the issue_1 branch.
This pull request should fix #1.

### CODE REVIEW

**Comment by trink** [comment]
If we are going to change `id` to an unsigned long long we should change it everywhere.

File: src/hs_input.c [hunk 1]
```diff
@@ -183,9 +183,9 @@ void hs_free_input_buffer(hs_input_buffer* b)
  *
  * @return size_t Least power of 2 greater than or equal to x
  */
-static size_t clp2(size_t x)
+static unsigned long long clp2(unsigned long long x)
```
Comment (trink) [inline src/hs_input.c]: This should remain size_t since it is used as input to malloc.

File: src/hs_output.h [hunk 2]
```diff
@@ -18,7 +18,7 @@ typedef struct hs_output
   FILE* fh;
   char* path;
   pthread_mutex_t lock;
-  size_t id;
```
Comment (trink) [inline src/hs_output.h]: This should be replaced with hs_checkpoint (I missed it we reworking the checkpoints)

**Comment by bbinet** [comment]
I've rework my patch so that we don't need to change `size_t` to `unsigned long long` type.

File: src/hs_output.h (hunk 2 above)
Comment (bbinet) [inline src/hs_output.h]: my new patch doesn't change `hs_output.h` anymore, so I won't address this comment myself in this PR

**Comment by trink** [comment]
let me just fix up the id the unsigned long long

**Comment by bbinet** [comment]
ok, thanks.

**Pull request closed without merging**